shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm

```diff
@@ -1965,13 +1966,26 @@ - (void)updateEditingStateWithDelta:(flutter::TextEditingDelta)delta {
     @"composingExtent" : @(composingExtent),
   };
 
-  NSDictionary* deltas = @{
-    @"deltas" : @[ deltaToFramework ],
-  };
+  [_pendingDeltas addObject:deltaToFramework];
 
-  [self.textInputDelegate flutterTextInputView:self
-                           updateEditingClient:_textInputClient
-                                     withDelta:deltas];
+  if (_pendingDeltas.count == 1) {
+    CFStringRef runLoopMode = self.customRunLoopMode != nil
+                                  ? (__bridge CFStringRef)self.customRunLoopMode
+                                  : kCFRunLoopCommonModes;
+
+    CFRunLoopPerformBlock(CFRunLoopGetMain(), runLoopMode, ^{
+      if (_pendingDeltas.count > 0) {
+        NSDictionary* deltas = @{
+          @"deltas" : _pendingDeltas,
+        };
+
+        [self.textInputDelegate flutterTextInputView:self
+                                 updateEditingClient:_textInputClient
+                                           withDelta:deltas];
```

Comment:
I think it is fine but we should confirm that the framework can handle any delta batches. cc @LongCatIsLooong

Reply:
Are you referring to the frameworks ability to receive/apply delta batches?